src/app/Library/CrudPanel/Traits/Tabs.php

```diff
@@ -106,56 +106,77 @@ public function isLastTab($label)
     }
 
     /**
+     * @deprecated Do not use this method as it will be removed in future versions!
+     * Instead, use $this->getElementsWithoutATab($this->getCurrentFields())
+     *
      * @return \Illuminate\Support\Collection
      */
     public function getFieldsWithoutATab()
     {
-        $all_fields = $this->getCurrentFields();
+        return $this->getElementsWithoutATab($this->getCurrentFields());
+    }
 
-        $fields_without_a_tab = collect($all_fields)->filter(function ($value, $key) {
+    /**
+     * @return \Illuminate\Support\Collection
+     */
+    public function getElementsWithoutATab(array $elements)
+    {
+        return collect($elements)->filter(function ($value) {
             return ! isset($value['tab']);
         });
+    }
 
-        return $fields_without_a_tab;
+    /**
+     * @return array|\Illuminate\Support\Collection
+     */
+    public function getTabFields(string $label)
+    {
+        return $this->getTabElements($label, $this->getCurrentFields());
     }
 
     /**
-     * @param $label
      * @return array|\Illuminate\Support\Collection
      */
-    public function getTabFields($label)
+    public function getTabColumns(string $label)
     {
-        if ($this->tabExists($label)) {
-            $all_fields = $this->getCurrentFields();
+        return $this->getTabElements($label, $this->columns());
+    }
 
-            $fields_for_current_tab = collect($all_fields)->filter(function ($value, $key) use ($label) {
+    /**
+     * @return array|\Illuminate\Support\Collection
+     */
+    public function getTabElements(string $label, array $elements)
+    {
+        if ($this->tabExists($label)) {
+            return collect($elements)->filter(function ($value) use ($label) {
                 return isset($value['tab']) && $value['tab'] == $label;
             });
-
-            return $fields_for_current_tab;
         }
 
         return [];
     }
 
-    /**
-     * @return array
-     */
-    public function getTabs()
+    public function getTabs(): array
     {
         $tabs = [];
-        $fields = $this->getCurrentFields();
 
-        $fields_with_tabs = collect($fields)
-            ->filter(function ($value, $key) {
+        collect($this->getCurrentFieldsOrColumns())
+            ->filter(function ($value) {
                 return isset($value['tab']);
             })
-            ->each(function ($value, $key) use (&$tabs) {
+            ->each(function ($value) use (&$tabs) {
                 if (! in_array($value['tab'], $tabs)) {
                     $tabs[] = $value['tab'];
                 }
             });
 
         return $tabs;
     }
+
+    public function getCurrentFieldsOrColumns(): array
```

Comment:
We should re-think this function as this would limit the tabs functionality. Either `show => columms`, anything else `fields`. 

I think to make this "operation free", so any operation can have the tabs, we need to find a way that this function can get the "elements" (fields/columns etc).

At a first glance, I think if we introduced something like: `operationElements` setting (i am still not sold on `elements` keyword 🟡 :
```php
// create/update
CRUD::setOperationSetting('operationElements', function($crud) { return $crud->fields(); } )

// list/show
CRUD::setOperationSetting('operationElements', function($crud) { return $crud->columns(); } )
```
I am concerned that this change could snowball into something very big, because https://github.com/Laravel-Backpack/CRUD/pull/4854 came to my mind after writing this, so it might not be a good idea to think about it now as it would be a change too broad.

I still don't like the fact that we are tying tabs to operations and I would like that bit to be operation responsibility rather `Tabs`. 

What if operations explicitly set the tab elements (or whatever, maybe `items` ?) instead of the "operation elements" as I thought at first?

```php
create/update
CRUD::getTabItemsFrom('fields')
// show
CRUD::getTabItemsFrom('columns')
```

This by default would call `$crud->{$itemsFrom}()`, but we should also probably allow a callback `CRUD::getTabItemsFrom(function($crud) { $crud->whatever(); })`

Let me know what you think! 👍 

Cheers